src/approaches/reinforcement_learning_approach.py

```diff
@@ -0,0 +1,92 @@
+                 initial_options: Set[ParameterizedOption], types: Set[Type],
+                 action_space: Box, train_tasks: List[Task]) -> None:
+        super().__init__(initial_predicates, initial_options, types,
+                         action_space, train_tasks)
+        self._nsrts: Set[NSRT] = set()
+        self._online_learning_cycle = 0
+        self._initial_trajectories: List[LowLevelTrajectory] = []
+        self._online_trajectories: List[LowLevelTrajectory] = []
+        # TODO: initialize option learner
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "nsrt_rl"
+
+    def learn_from_offline_dataset(self, dataset: Dataset) -> None:
+        self._initial_trajectories = dataset.trajectories
+        super().learn_from_offline_dataset(dataset)
+
+    @classmethod
+    def _make_termination_fn(cls, goal: Set[GroundAtom]) \
+            -> Callable[[State], bool]:
+
+        def _termination_fn(s: State) -> bool:
+            return all(goal_atom.holds(s) for goal_atom in goal)
+
+        return _termination_fn
+
+    def get_interaction_requests(self) -> List[InteractionRequest]:
+        # For each training task, try to solve the task to get a policy. If the
+        # task can't be solved, construct a policy from the sequence of _Options
+        # that achieves the longest partial refinement of a valid plan skeleton.
+        # The teacher will collect a trajectory on the training task using this
+        # policy.
+        requests = []
+        for i in range(len(self._train_tasks)):
+            task = self._train_tasks[i]
+            try:
+                _act_policy = self.solve(task, CFG.timeout)
+            except (ApproachTimeout, ApproachFailure) as e:
+                partial_refinements = e.info.get("partial_refinements")
+                assert partial_refinements is not None
+                _, plan = max(partial_refinements, key=lambda x: len(x[1]))
+                _act_policy = utils.option_plan_to_policy(plan)
+            request = InteractionRequest(
+                train_task_idx=i,
+                act_policy=_act_policy,
+                query_policy=lambda s: None,
+                termination_function=self._make_termination_fn(task.goal))
+            requests.append(request)
+        return requests
+
+    def learn_from_interaction_results(
+            self, results: Sequence[InteractionResult]) -> None:
+        self._online_learning_cycle += 1
+        # We get one result per training task.
+        for i, result in enumerate(results):
+            states = result.states
+            actions = result.actions
+            traj = LowLevelTrajectory(states,
+                                      actions,
+                                      _is_demo=False,
+                                      _train_task_idx=i)
+            self._online_trajectories.append(traj)
+
+        # Replace this with an _RLOptionLearner.
+        self._learn_nsrts(self._initial_trajectories + \
```

Comment:
oh i see, so we're not actually learning options right now, this is like a placeholder?

Reply:
So, we got more data after interacting with the environment during our online learning cycle. Now we get to choose how we want to update our options based on the new data we got. We want `reinforcement_learning_approach.py` to implement various RL based ways to update these options. 

So yeah– this line is currently a placeholder, where we update the options in a different manner, that is, doing behavioral cloning all over again, on the total dataset we have so far.